src/smd-core.c

```diff
@@ -982,7 +982,10 @@ static int smd_attr_copy_val_to_internal(char *out, smd_dtype_t *t, smd_dtype_t
         case (SMD_TYPE_INT8): {
           int8_t ov = *(int8_t*)val;
           // check the range
-          if(ov < FLT_MIN || ov > FLT_MAX){
+          if(ov > 0 && (ov < FLT_MIN || ov > FLT_MAX)){
+            return 1;
+          }
+          if(ov < 0 && (ov > -FLT_MIN || ov < -FLT_MAX)){
```

Comment:
`int8_t` cannot be outside of the range of `float`, checking for it is pointless. All that this check does is to add clutter to the code. The same goes for the `int16_t` to `float` conversion. The `float` type has 24 effective bits, and as long as your source type has less than that, all values are exactly representable. Likewise, the `double` type has 53 effective bits, and any integer type with less than 53 bits is exactly representable by it.

This is a systematic problem, and I won't add comments to all the conversions to `float` and `double` as they are analogous.

Reply:
There are extra tests and it's intentional for now. We can remove the ones that don't make sense later. I implemented all the combinations.